src/main/java/site/youtogether/resolver/AddressArgumentResolver.java

```diff
@@ -0,0 +1,32 @@
+package site.youtogether.resolver;
+
+import org.springframework.core.MethodParameter;
+import org.springframework.stereotype.Component;
+import org.springframework.web.bind.support.WebDataBinderFactory;
+import org.springframework.web.context.request.NativeWebRequest;
+import org.springframework.web.method.support.HandlerMethodArgumentResolver;
+import org.springframework.web.method.support.ModelAndViewContainer;
+
+import jakarta.servlet.http.HttpServletRequest;
+
+@Component
+public class AddressArgumentResolver implements HandlerMethodArgumentResolver {
+
+	@Override
+	public boolean supportsParameter(MethodParameter parameter) {
+		boolean hasAddressAnnotation = parameter.hasParameterAnnotation(Address.class);
+		boolean isStringType = String.class.isAssignableFrom(parameter.getParameterType());
+
+		return hasAddressAnnotation && isStringType;
+	}
+
+	@Override
+	public Object resolveArgument(MethodParameter parameter, ModelAndViewContainer mavContainer, NativeWebRequest webRequest,
+		WebDataBinderFactory binderFactory) throws Exception {
+		HttpServletRequest request = webRequest.getNativeRequest(HttpServletRequest.class);
+
+		assert request != null;
```

Comment:
assert 를 사용하면 얻는 효과가 뭔가요?

Reply:
초기 단계에서 객체를 확인을 하여 나중에 더 모호한 NullPointerException으로 실패하는 것을 방지합니다.
위 코드는 IDE에서 제안하는 사항 중 하나입니다.

좀 더 개선된 응답을 제공하고자 한다면, 다음과 같이 작성할 수 있습니다.
```suggestion
		if (request == null) {
		    throw new IllegalArgumentException("HttpServletRequest는 null일 수 없습니다");
		}
```